### ISSUE TITLE
Toil-scripts installed in venv causes Toil resolveEntryPoint error

### ISSUE THREAD

**Issue by jvivian** [BD2KGenomics/toil#929]
Trying to get pip installable toil-scripts to work in a venv causes a Toil error. Any cgcloud box with toil installed will work, a toil cluster isn't needed. 

To reproduce:

**Laptop**
```
cgcloud create-cluster --instance-type m3.large --leader-instance-type m3.large --spot-bid 1.00 --num-workers 1 toil

cgcloud ssh toil-leader
```

**Leader**
```
virtualenv --system-site-packages piptest

source piptest/bin/activate

pip install --pre toil-scripts
-----------------------------------
Collecting toil-scripts
Requirement already satisfied (use --upgrade to upgrade): boto==2.38.0 in /usr/local/lib/python2.7/dist-packages (from toil-scripts)
Requirement already satisfied (use --upgrade to upgrade): toil==3.1.4 in /usr/local/lib/python2.7/dist-packages (from toil-scripts)
Collecting pyyaml==3.11 (from toil-scripts)
Collecting tqdm==3.8.0 (from toil-scripts)
  Using cached tqdm-3.8.0-py2.py3-none-any.whl
Requirement already satisfied (use --upgrade to upgrade): bd2k-python-lib==1.10.dev6 in /usr/local/lib/python2.7/dist-packages (from toil==3.1.4->toil-scripts)
Installing collected packages: pyyaml, tqdm, toil-scripts
  Found existing installation: PyYAML 3.10
    Not uninstalling pyyaml at /usr/lib/python2.7/dist-packages, outside environment /home/mesosbox/piptest
Successfully installed pyyaml-3.11 toil-scripts-2.0a2.dev179 tqdm-3.8.0
```

While pip says the requirement is already satisfied, something is wrong with the installation as Toil complains that it can't find an entrypoint:

```
INFO:toil.leader:Starting the main loop
Traceback (most recent call last):
  File "./piptest/bin/toil-rnaseq", line 11, in <module>
    sys.exit(main())
  File "/home/mesosbox/piptest/local/lib/python2.7/site-packages/toil_scripts/rnaseq_cgl/rnaseq_cgl_pipeline.py", line 638, in main
    Job.Runner.startToil(Job.wrapJobFn(map_job, download_sample, samples, config), args)
  File "/usr/local/lib/python2.7/dist-packages/toil/job.py", line 445, in startToil
    return mainLoop(config, batchSystem, jobStore, rootJob, jobCache=jobCache)
  File "/usr/local/lib/python2.7/dist-packages/toil/leader.py", line 392, in mainLoop
    jobWrapper.cores, jobWrapper.disk)
  File "/usr/local/lib/python2.7/dist-packages/toil/leader.py", line 101, in issueJob
    jobCommand = ' '.join((resolveEntryPoint('_toil_worker'), self.jobStoreString, jobStoreID))
  File "/usr/local/lib/python2.7/dist-packages/toil/__init__.py", line 45, in resolveEntryPoint
    assert os.path.isfile(path)
AssertionError
```

This error *doesn't* appear and the pipeline can run locally if toil-scripts is installed outside of a venv.

**Comment by hannes-ucsc** [comment]
Excellent description.

**Comment by hannes-ucsc** [comment]
I'm assuming the image was created from toil-box, not toil-latest-box?

**Comment by hannes-ucsc** [comment]
Toil detects that a virtualenv is active and assumes that _toil_worker is present in that virtualenv's bin. It asserts that assumption and the assertion fails.

We currently distinguish two cases: 1) no virtualenv with `_toil_worker` on the PATH and 2) a virtualenv with `_toil_worker` in its bin directory. There needs to be a third case for virtualenvs that have the Toil packages on the `sys.path` but whose bin directory lacks the `_toil_worker` entry point. The only way to create such a virtualenv is with `--system-site-packages`.

**Comment by hannes-ucsc** [comment]
I don't think I want to backport this fix to 3.1.4. The workaround for that version should be to

```
pip install --pre --user toil-scripts
export PATH=~/.local/bin:$PATH
```

**Comment by hannes-ucsc** [comment]
For 3.2.0 we will add support for virtualenvs with `--system-site-packages`. I just need to decide if we assume that the virtualenv is present on each worker. If we make that assumption, we would open the road to user distributions (a pip-installable distribution containing Toil user modules, like toil-scripts) that have dependencies. OTOH, this would have the downside of requiring the user to place their virtualenv on every worker at exactly the same location.
